src/test/java/org/candlepin/subscriptions/tally/job/CaptureSnapshotsTaskManagerTest.java

```diff
@@ -145,6 +168,26 @@ private TaskDescriptor createDescriptor(List<String> accounts) {
             .build();
     }
 
+    protected OffsetDateTime adjustTimeForLatency(OffsetDateTime dateTime, Duration adjustmentAmount) {
+        return dateTime.toZonedDateTime().minus(adjustmentAmount).toOffsetDateTime();
+    }
+
+    private TaskDescriptor createHourlyDescriptor(String accountNumber) {
+        Duration metricRange = appProperties.getMetricLookupRangeDuration();
+        Duration prometheusLatencyDuration = appProperties.getPrometheusLatencyDuration();
+        Duration hourlyTallyOffsetMinutes = appProperties.getHourlyTallyOffsetMinutes();
+
+        OffsetDateTime endDateTime = adjustTimeForLatency(OffsetDateTime.now().minus(hourlyTallyOffsetMinutes)
+            .truncatedTo(ChronoUnit.HOURS), prometheusLatencyDuration);
+        OffsetDateTime startDateTime = endDateTime.minus(metricRange).minus(hourlyTallyOffsetMinutes);
+
+        return TaskDescriptor.builder(TaskType.UPDATE_HOURLY_SNAPSHOTS, taskQueueProperties.getTopic())
+            .setSingleValuedArg("accountNumber", accountNumber)
+            .setSingleValuedArg("startDateTime", startDateTime.toString())
+            .setSingleValuedArg("endDateTime", endDateTime.toString())
+            .build();
+    }
+
```

Comment:
Rather than repeat the logic from the `CaptureSnapshotsTaskManager`, I'd rather see real numbers in here. Specifically, I'd rather see this method take `startDateTime` and `endDateTime` as parameters. Then the test will ensure that we get the expected values for `startDateTime` and `endDateTime`.

It would help to use `@ContextConfiguration(classes = FixedClockConfiguration.class)`

Note this will actually require a tiny tweak to `CaptureSnapshotsTaskManager`: you'll need to inject the `ApplicationClock` and change `OffsetDateTime.now()` to `OffsetDateTime.now(applicationClock.getClock())` (as seen in `src/main/java/org/candlepin/subscriptions/retention/TallyRetentionPolicy.java`.

Reply:
One concern I have about this is this change puts this test out of sync with the way other tests are done, and with the way the CaptureSnapshotsTaskmanager.update... method is implemented.  I don't like copy / pasting code either, sure.  How literally does the test need to mimic / emulate the way the tested method works?

Reply:
I thought the test already checked the values.  The error I saw while sorting out the test suggests this.